### PR TITLE
sdk/node: add access control methods

### PR DESCRIPTION
**Pull request by dominic**

### CODE REVIEW

File: sdk/node/src/api/accessControl.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * More info: {@link https://chain.com/docs/core/learn-more/authentication}
+ * @typedef {Object} Grant
+ * @global
+ *
+ * @property {String} guard_type
+ * Type of credential, either 'access_token' or 'x509'.
+ *
+ * @property {Object} guard_data
+ * Object containing data needed to identify the incoming credential.
```
Comment (jeffomatic) [inline sdk/node/src/api/accessControl.js]: Folks will probably need a bit more hinting here, e.g.:

```
Data used by the guard to identity incoming credentials.

If guard_type is "access_token", you should provide an instance of {AccessTokenGuardData}, which identifies access tokens by ID.

If guard_type is "x509", you should...
```

File: sdk/node/src/api/accessControl.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const accessControl = (client) => ({
+  /**
+   * Create a new access grant.
+   *
+   * @param {Object} params - Parameters for access grant creation.
+   * @param {String} params.guard_type - Type of credential to guard with, either 'access_token' or 'x509'.
+   * @param {Object} params.guard_data - Object containing data needed to identify the incoming credential.
+   * @param {String} params.policy - Authorization polciy to attach to specific grant. See {@link Grant} for a list of available policiies.
+   * @param {objectCallback} [callback] - Optional callback. Use instead of Promise return value as desired.
+   * @returns {Promise} Status of created object.
```
Comment (jeffomatic) [inline sdk/node/src/api/accessControl.js]: Maybe s/Status of created object/Success message/

File: sdk/node/src/api/accessControl.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  create: (params , cb) =>
+    shared.create(client, '/create-acl-grant', params, {skipArray: true}, cb),
```
Comment (jeffomatic) [inline sdk/node/src/api/accessControl.js]: I think `cb` should be a field in the options object. Have you run this code yet?

File: sdk/node/src/api/accessControl.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    cb
+  ),
+
+  /**
+   * Get all access grants.
+   *
+   * @param {pageCallback} [callback] - Optional callback. Use instead of Promise return value as desired.
+   * @returns {Promise<Array<Grant>>} Requested page of results.
+   */
+  query: (cb) =>
```
Comment (jeffomatic) [inline sdk/node/src/api/accessControl.js]: Should there be a `queryAll` equivalent? Or did we get rid of pagination?

File: sdk/node/src/api/accessControl.js (hunk 2 above)
Comment (dominic) [inline sdk/node/src/api/accessControl.js]: We use this language elsewhere; should I do an everything update here?

File: sdk/node/src/api/accessControl.js (hunk 3 above)
Comment (dominic) [inline sdk/node/src/api/accessControl.js]: I've been running the promise version; good catch.

File: sdk/node/src/api/accessControl.js (hunk 4 above)
Comment (dominic) [inline sdk/node/src/api/accessControl.js]: Surprisingly, my vote is for no pagination right now.

File: sdk/node/src/api/accessControl.js (hunk 2 above)
Comment (dominic) [inline sdk/node/src/api/accessControl.js]: Actually, scratch that. I totally updated markdown in this branch too.

**Comment by dominic** [comment]
@jeffomatic PTAL again.

re: integration tests, yes. There are no APIs landed in core yet. I'm happy to hold this back until we have them though.

**Comment by dominic** [comment]
Feedback from Jeff: let's rename `query` to `list`.

**Comment by dominic** [comment]
@jeffomatic Updated - PTAL again

**Comment by jeffomatic** [comment]
LGTM but would you mind waiting until we get tests?

File: sdk/node/src/api/mockHsmKeys.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ const shared = require('../shared')
   * Cryptographic private keys are the primary authorization mechanism on a
   * blockchain. For development environments, Chain Core provides a convenient
   * MockHSM.
-  * <br/><br/>
+  *
```
Comment (kr) [inline sdk/node/src/api/mockHsmKeys.js]: It looks to me like almost 25% of this PR is unrelated edits like this one. It would be nice to keep each PR focused on one thing when possible.

Comment (dominic) [inline sdk/node/src/api/mockHsmKeys.js]: Yup. In this case, changing https://github.com/chain/chain/pull/931/files#diff-84c1d078debc432e4577dbe04061249aR16 to support https://github.com/chain/chain/pull/931/files#diff-f138fcf4ae92f3121a6c5988001b5f6fR4 broke a bunch of formatting elsewhere. I can split it up easily enough if that's preferred.
